packages/lumx-react/src/components/input-label/InputLabel.tsx

```diff
@@ -44,14 +51,20 @@ const DEFAULT_PROPS: Partial<InputLabelProps> = {};
  */
 export const InputLabel = forwardRef<InputLabelProps, HTMLLabelElement>((props, ref) => {
     const defaultTheme = useTheme() || Theme.light;
-    const { children, className, htmlFor, isRequired, theme = defaultTheme, ...forwardedProps } = props;
+    const { children, className, htmlFor, isRequired, theme = defaultTheme, typography, ...forwardedProps } = props;
+    const typographyClass = typography && getTypographyClassName(typography);
 
     return (
         <label
             ref={ref}
             {...forwardedProps}
             htmlFor={htmlFor}
-            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, isRequired, theme }))}
+            className={classNames(
+                className,
+                handleBasicClasses({ prefix: CLASSNAME, isRequired, theme }),
+                getBasicClass({ prefix: CLASSNAME, type: 'hasCustomTypography', value: Boolean(typography) }),
```

Comment:
This should be simpler and work the same

```suggestion
                handleBasicClasses({ prefix: CLASSNAME, isRequired, theme, hasCustomTypography: Boolean(typography) }),
```